hydra/_internal/core_plugins/fish_completion.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+import os
+import re
+import sys
+from typing import List, Optional, Tuple
+
+from hydra.plugins.completion_plugin import CompletionPlugin
+
+log = logging.getLogger(__name__)
+
+
+class FishCompletion(CompletionPlugin):
+    """Issue 1: python script.py style rule cannot be uninstalled
+    Issue 2: fish adds a space to "hydra. " automatically"""
+
+    def install(self) -> None:
+        output = self._get_exec()
+        script = []
+        for name, cond, exe in output:
+            args = f"'({exe} -sc query=fish --_completion (commandline -cp))'"
+            script.append(f"""complete -c {name} {cond}-x -a {args}\n""")
+        print("".join(script))
+
+    def uninstall(self) -> None:
+        name = self._get_uninstall_exec()
+        print(f"complete -e -c {name}")
+
+    def provides(self) -> str:
+        return "fish"
+
+    @staticmethod
+    def strip_python_or_app_name(line: str) -> str:
+        """
+        Take the command line received from bash completion, and strip the app name from it
+        which could be at the form of python script.py or some_app.
+        it also corrects the key (COMP_INDEX) to reflect the same location in the striped command line.
+        :param line: input line, may contain python file.py followed=by_args..
+        :return: tuple(args line, key of cursor in args line)
+        """
+        python_args = r"^\s*[\w\/]*python[23]?\s*[\w/\.]*\s*(.*)"
+        app_args = r"^\s*[\w_\-=\./]+\s*(.*)"
+        match = re.match(python_args, line)
+        if match:
+            return match.group(1)
+        else:
+            match = re.match(app_args, line)
+            if match:
+                return match.group(1)
+            else:
+                raise RuntimeError("Error parsing line '{}'".format(line))
```

Comment:
I changed the regexp a bit to also match `python2` and `python3` instead of only `python` (just noticed python2 is not supported). I wasn't sure the consideration in the original version, but this change should be generic enough to be moved in the base class.